src/server/scripts/Outland/HellfireCitadel/HellfireRamparts/boss_vazruden_the_herald.cpp

```diff
@@ -96,7 +96,7 @@ class boss_nazan : public CreatureScript
             void IsSummonedBy(WorldObject* summoner) override
             {
                 if (summoner->GetEntry() == NPC_VAZRUDEN_HERALD)
-                    VazrudenGUID = summoner->GetGUID();
+                    Vazruden = me->FindNearestCreature(NPC_VAZRUDEN, 5000);
```

Comment:
what is the meaning of this change ? what issue is it trying to fix and how is the new code fixing it ?

Reply:
Wish I knew for certain, but could it be a distance issue, making the `summoner` not getting the GUID in the original code?

Reply:
@illfated that's mostly speculation. If I have to merge this PR, I want someone with some C++ and TC knowledge to have checked it before, otherwise I might just take it as suggestion and debug the issue on my own.

The only case where that code returns something different is if there are 2+ NPC_VAZRUDEN, otherwise summoner is already what me->FindNearestCreature(NPC_VAZRUDEN, 5000) returns.

Reply:
OK, very well, fair point. I worry that the OP does not know these details either.

Reply:
I would like to have this line reverted and know if the rest of the PR still solves any issue

Reply:
Yes, this change causes the build to fail, so it is not a solution.